asmcomp/arm64/proc.ml

```diff
@@ -109,6 +109,36 @@ let loc_spacetime_node_hole = Reg.dummy  (* Spacetime unsupported *)
 
 (* Calling conventions *)
 
+let loc_int last_int make_stack int ofs =
+  if !int <= last_int then begin
+    let l = phys_reg !int in
+    incr int; l
+  end else begin
+    ofs := Misc.align !ofs size_int;
+    let l = stack_slot (make_stack !ofs) Int in
+    ofs := !ofs + size_int; l
+  end
+
+let loc_float last_float make_stack float ofs =
+  if !float <= last_float then begin
+    let l = phys_reg !float in
+    incr float; l
+  end else begin
+    ofs := Misc.align !ofs size_float;
+    let l = stack_slot (make_stack !ofs) Float in
+    ofs := !ofs + size_float; l
+  end
+
+let loc_int32 last_int make_stack int ofs =
+  if !int <= last_int then begin
+    let l = phys_reg !int in
+    incr int; l
+  end else begin
+    let l = stack_slot (make_stack !ofs) Int in
+    ofs := !ofs + (if macosx then 4 else 8);
```

Comment:
Hmm wouldn't it be a better idea to put it in a variable?

`let size_int32 = if macosx then 4 else size_int`

Reply:
Not with the name `size_int32`, which for me means a solid "4" and nothing else.  Possibly with the name `size_int32_argument_on_stack`.  But I prefer the original `if macosx then 4 else 8`, it's documentation in code ("under macOS this stack slot spans 4 bytes, otherwise it spans 8 bytes").